### PR TITLE
doc: Rework generate* doc

### PR DESCRIPTION
**Pull request by maflcko**
Hide the test-only calls and clarify the short description

### CODE REVIEW

**Comment by MarcoFalke** [comment]
Rendered diff:

```diff
diff --git a/generateblock b/generateblock
index 55aca89..1fcd557 100644
--- a/generateblock
+++ b/generateblock
@@ -1,6 +1,6 @@
 generateblock "output" ["rawtx/txid",...]
 
-Mine a block with a set of ordered transactions immediately to a specified address or descriptor (before the RPC call returns)
+Mine a set of ordered transactions to a specified address or descriptor and return the block hash.
 
 Arguments:
 1. output               (string, required) The address or descriptor to send the newly generated bitcoin to.
diff --git a/generatetoaddress b/generatetoaddress
index 04c2092..9083a40 100644
--- a/generatetoaddress
+++ b/generatetoaddress
@@ -1,9 +1,9 @@
 generatetoaddress nblocks "address" ( maxtries )
 
-Mine blocks immediately to a specified address (before the RPC call returns)
+Mine to a specified descriptor and return the block hashes.
 
 Arguments:
-1. nblocks     (numeric, required) How many blocks are generated immediately.
+1. nblocks     (numeric, required) How many blocks are generated.
 2. address     (string, required) The address to send the newly generated bitcoin to.
 3. maxtries    (numeric, optional, default=1000000) How many iterations to try.
 
diff --git a/generatetodescriptor b/generatetodescriptor
index cfbcf86..9572971 100644
--- a/generatetodescriptor
+++ b/generatetodescriptor
@@ -1,9 +1,9 @@
 generatetodescriptor num_blocks "descriptor" ( maxtries )
 
-Mine blocks immediately to a specified descriptor (before the RPC call returns)
+Mine to a specified descriptor and return the block hashes.
 
 Arguments:
-1. num_blocks    (numeric, required) How many blocks are generated immediately.
+1. num_blocks    (numeric, required) How many blocks are generated.
 2. descriptor    (string, required) The descriptor to send the newly generated bitcoin to.
 3. maxtries      (numeric, optional, default=1000000) How many iterations to try.

**Review by luke-jr — CHANGES_REQUESTED** [review]
Concept ACK, but I think it should keep the explicit mention of not returning until the blocks are mined.

**Comment by MarcoFalke** [comment]
It should be common sense that a hash of something can only be returned after it has been created (mined), so I am leaving this as-is for now

**Comment by MarcoFalke** [comment]
Also, the wording "immediately" isn't true. On testnets with difficulty higher than regtest, it may take a few iterations.

File: src/rpc/mining.cpp [hunk 1]
```diff
@@ -261,18 +261,18 @@ static RPCHelpMan generate()
...
-                    {"address", RPCArg::Type::STR, RPCArg::Optional::NO, "The address to send the newly generated bitcoin to."},
-                    {"maxtries", RPCArg::Type::NUM, RPCArg::Default{DEFAULT_MAX_TRIES}, "How many iterations to try."},
-                },
-                RPCResult{
-                    RPCResult::Type::ARR, "", "hashes of blocks generated",
-                    {
-                        {RPCResult::Type::STR_HEX, "", "blockhash"},
-                    }},
-                RPCExamples{
+        "Mine to a specified descriptor and return the block hashes.",
```
Comment (achow101) [inline src/rpc/mining.cpp:264]: s/descriptor/address

```suggestion
        "Mine to a specified address and return the block hashes.",
```

File: src/rpc/mining.cpp [hunk 2]
```diff
@@ -302,7 +302,7 @@ static RPCHelpMan generatetoaddress()
 static RPCHelpMan generateblock()
 {
     return RPCHelpMan{"generateblock",
-        "\nMine a block with a set of ordered transactions immediately to a specified address or descriptor (before the RPC call returns)\n",
```
Comment (0xB10C) [inline src/rpc/mining.cpp:305]: I'm not familiar with RPCHelpMan. Why can we drop the `\n` here and at the two other occurrences?

**Review by 0xB10C — COMMENTED** [review]
Code Review ACK fa3bb584dcc742a767b2141cd7324877e3cf5302

Only reviewed changes to the RPC docs via https://github.com/bitcoin/bitcoin/pull/24137#issuecomment-1019976927. 

`bitcoin-cli -help` lists `-generate` and says that it mines the blocks _immediately_. Could be changed there too.

```
  -generate
       Generate blocks immediately, equivalent to RPC getnewaddress followed by
       RPC generatetoaddress. Optional positional integer arguments are
       number of blocks to generate (default: 1) and maximum iterations
       to try (default: 1000000), equivalent to RPC generatetoaddress
       nblocks and maxtries arguments. Example: bitcoin-cli -generate 4
       1000
```

File: src/rpc/mining.cpp (hunk 2 above)
Comment (MarcoFalke) [reply src/rpc/mining.cpp:305]: Starting and trailing whitespace is normalized by `RPCHelpMan`. See also the rendered diff: https://github.com/bitcoin/bitcoin/pull/24137#issuecomment-1019976927

**Comment by MarcoFalke** [comment]
Thanks, rebased and fixed up bitcoin-cli.cpp. Should be trivial to re-review with `git range-diff bitcoin-core/master fa3bb584dc fa68953679`.

**Comment by 0xB10C** [comment]
reACK fa30e62cc609ff29f0acaa5047d3f437cb04a67b. changes since fa3bb584dcc742a767b2141cd7324877e3cf5302 are: dropping the `immediately` + formatting the touched line and a rebase

**Pull request merged**